### PR TITLE
Update SAR Extension

### PR DESCRIPTION
**Pull request by duckontheweb**
**Related Issue(s):**

- #326
- #327
- #262

**Description:**

- Updates docstrings for SAR Extension implementation
- Adds Collection summaries for SAR Extension
- Adds migration for pre-0.9 SAR fields:
    - `sar:type` -> `sar:product_type`
    - `sar:polarization` -> `sar:polarizations`

**PR Checklist:**

- [x] Code is formatted (run `pre-commit run --all-files`)
- [x] Tests pass (run `scripts/test`)
- [x] Documentation has been updated to reflect changes, if applicable
- [x] This PR maintains or improves overall codebase code coverage.
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/pystac/blob/develop/CHANGELOG.md). See [the docs](https://pystac.readthedocs.io/en/latest/contributing.html#changelog) for information about adding to the changelog.

### CODE REVIEW

File: pystac/extensions/sar.py [hunk 1]
```diff
@@ -353,27 +567,45 @@ def migrate(
...
+                    PREFIX + "type"
+                )
+
+            if (
+                PREFIX + "polarization" in obj["properties"]
+                and POLARIZATIONS_PROP not in obj["properties"]
+            ):
+                obj["properties"][POLARIZATIONS_PROP] = [
+                    obj["properties"].pop(PREFIX + "polarization")
                 ]
```
Comment (duckontheweb) [inline pystac/extensions/sar.py:608]: @m-mohr Can you verify that this is the right way to handle these migrations for the changes in v0.9.0?

Comment (m-mohr) [reply pystac/extensions/sar.py:608]: I'm not 100% fluent in Python. sar:constellation/instrument(s)/platform is looking good.

What is this the code for type and polarization(s) doing?

Comment (duckontheweb) [reply pystac/extensions/sar.py:608]: For `sar:type` this is converting the `sar:type` property to a `sar:product_type` property. For `sar:polarization` it takes the `sar:polarization` value and puts it into a `sar:polarizations` list.

So this:

```json
{
    "sar:type": "SSC",
    "sar:polarization": "HH"
}
```

would be converted to this:

```json
{
    "sar:product_type": "SSC",
    "sar:polarizations": ["HH"]
}
```

Comment (m-mohr) [reply pystac/extensions/sar.py:608]: That looks good to me.

**Review by lossyrob — APPROVED** [review]

**Pull request merged**